### ISSUE TITLE
Window flickering during resize

### ISSUE THREAD

**Issue by friedrich** [alacritty/alacritty#7898]
The client area of the window intermittently becomes transparent during resize, which causes flickering. See video. Possibly related to https://github.com/alacritty/alacritty/issues/969.

https://github.com/alacritty/alacritty/assets/107749/2479a89c-3ab2-4c31-8988-a523e838aea9

I explored some ways to fix the problem, but am uncertain on how to proceed. Any one of the following code changes fixes or strongly reduces flickering:
- Enable vsync by removing the call to `surface.set_swap_interval` in `alacritty/src/display/mod.rs`
- Sleep for 10 ms right after the call to `self.surface.resize` in `alacritty/src/display/mod.rs`
- Call `window_context.draw` during `WinitEvent::AboutToWait` instead of `WindowEvent::RedrawRequested` in `alacritty/src/event.rs`

Changing `.with_transparent(true)` to `.with_transparent(false)` in `alacritty/src/display/window.rs` helps a bit, since the window does not become transparent, bit it still flickers, exposing the system window color instead.

### System

OS: macOS 14.4.1
Version: alacritty 0.13.2 (bb8ea18), alacritty 0.14.0-dev (d4f2f857)

**Comment by kchibisov** [comment]
Is example from https://github.com/rust-windowing/winit flickers? Also, is example from github.com/rust-windowing/glutin/ flickers?

**Comment by friedrich** [comment]
No, they don't. The problem seems to be related to glutin, but the glutin example has vsync enabled and also renders in `AboutToWait`.

If I change the glutin example such that vsync is disabled and if I move the render call to `RedrawRequested`, it flickers exactly like alacritty does.

**Comment by kchibisov** [comment]
I feel like it's a macOS sync issue or something like that. Because the `RedrawRequested` is essentially a `drawRect` during resize, well, more or less, so if it flickers it could be some sync issue on apple side?

And with e.g. `AboutToWait` it's when events are cleared, thus it likely stretches the content because the resize is not issued at the `drawRect`, but then  we submit the right buffer so it looks ok-ish.
